### DISCUSSION TITLE
Add in the thief and Hiester roles.

### DISCUSSION THREAD

**Discussion by MaxIsJoe** [unitystation/unitystation#8933] [Ideas]
Thief and hiester antags are different from regular antags because they do not have objectives; They instead have a score of how many valuables they have at the end of the round.

Stealing is the entire point of their role, they're not designed for murder, nor they are equipped with any way to obtain tools from a syndicate shop or anything like that.
They must make use of the station they're in to create tools and opportunities that lets them get into places without getting caught or getting violent.
Their win condition is based around getting high scores, with thieves competing against previous round theives' own scores in a leaderboard and the player's own all-time high score.

Hiesters spawn midround in the middle of all the chaos to compete against the starting thieves. They gain a few tools to mask their identity and gain access around the station quickly as a head start, Their win condition is to get a higher score than the other thieves.

## Design as discussed on Discord:

- Thieves have a win condition of getting a high score by stealing valuables from the station.
- Thieves receive a penalty whenever they attack any other player no matter what.
- Thieves must reach the end of the round without being in handcuffs. (aka getting caught)
- Thieves start with no equipment and must make use of the station's own resources to help them do their goal.
- Thieves that join mid-round whenever there are more than 18 players become hiesters who compete against other Thieves.
- Hiesters must earn a higher score than the other thieves to win and all other conditions are applied to them as well.
- Hiesters start with some equipment that gives them a head start.

## Feature Requirements for these roles:

- [ ] A way to hide your identity.
- [ ] Balancing all items to have price tags.
- [x] A modular score tracker rather than an objective tracker. Can be re-used in other instances like station score.
- [ ] Bag of holding.
- [ ] High score history (Leaderboards and player high scores).
- [ ] The ability to withdraw cargo credits as physical cash.
- [ ] Admin event / ghost role.

**Comment by MaxIsJoe** [comment]
I believe these two antags are incredibly easy to add and they only need some much needed improvements to systems we already have.
We can take down two birds with one stone if we focus on some these requirements i outlined above since they're very small.

- A bag of holding is just an item container with large storage space and with a funny interaction if merged together
- identiy concelement is something we needed to add anyways to have some parity with /tg/
- we need score trackers for stuff, not just these antags. a modular system that lets us keep track of stuff at the end of the round and asses a score for it will be great for things like station objectives/score.
- Highscore history can be done when we have the django stuff sorted but we can still locally save it on the server's disk in a json file in the meantime
- We needed a way to give antag to players without respawning them anyways, this is a must have regardless if we add these roles or not.

**Comment by PerfectTangent** [comment]
Firstly, it seems like a good simple antag, like acquire magic antag, with a clear goal that refuses established game mechanics. Would be a good addition.

I don't see any reason to have a distinction between the two. They aren't different in a meaningful way. The difference stated, one being round start and the other being midround, is not in of itself grounds for them to be separated. We don't call a mid round nukie as something different from from a round start nukie; they are the same role with the same objectives.

You do describe giving the opportunist tools that the thief doesn't get, but is that enough for them to be separate when it's a toolbox and a perhaps a bag of holding? In /tg/, the fugitive comes in several quite different flavors, but they're all still fugitives because their goal hasn't changed.

another thought, why does the opportunist get extra gear? I'm not necessarily against it. I just wonder what your thinking was. Why do they get it when a thief doesn't? Why do they have the gear if they are an _opportunist_ who is taking an opportunity in the moment? 

Being handed the tools to expedited I think might take away some of the variety that this antag could potentially have. Figuring out how to work with what's around you is a stated goal, so giving them gear seems counterintuitive to that.

Separate note about the cash withdrawal, Aaron made IDs able to track currencies a while ago. That seems like a  sufficient method to me to track stolen credits.

**Reply by MaxIsJoe** [reply]
> another thought, why does the opportunist get extra gear?

From my understanding on why it's done like this in ss13 is because the oppritunust joins waay too late into the round that obtaining tools to steal a lot of stuff becomes harder and slower than the thief who had a couple of minutes to prepare at round start.
Oppritunusts do not obtain a bag of holding on spawn, they just have a toolbox that lets them quickly test out door hacking and get around the station.

**Reply by PerfectTangent** [reply]
why's the bag of holding mentioned, then? It'd be nice to have, most definitely, but why under this list?

**Reply by MaxIsJoe** [reply]
> why's the bag of holding mentioned, then? It'd be nice to have, most definitely, but why under this list?

Its the only item storage container that can hold large amounts of items. Currenlty the normal backpack and by extension a lot of other item storage methods are not efficent and can limit how many items you can hold. 
Thiefs usually try getting a hold of a bag of holding to make their lives easier when stealing stuff without worrying too much about storage.

**Comment by MaxIsJoe** [comment]
Updated based on discussions we had on discord and here.
